src/TestFramework/TestFramework/Attributes/TestMethod/RetryAttribute.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestTools.UnitTesting;
+
+/// <summary>
+/// This attribute is used to set a retry count on a test method in case of failure.
+/// </summary>
+[AttributeUsage(AttributeTargets.Method, Inherited = false, AllowMultiple = false)]
+public sealed class RetryAttribute : Attribute
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RetryAttribute"/> class with the given number of max retries.
+    /// </summary>
+    public RetryAttribute(int maxRetries)
```

Comment:
I don't want to build something too complex but I think it would be good to have:

- Delay: wait time between 2 retries
- optional: some backoff strategy, I think constant and exponential are the most used.